src/envoy/auth/config.h

```diff
@@ -38,21 +38,25 @@ class AsyncClientCallbacks : public AsyncClient::Callbacks,
  public:
   AsyncClientCallbacks(Upstream::ClusterManager &cm, const std::string &cluster,
                        std::function<void(bool, const std::string &)> cb)
-      : cm_(cm),
+      : cancel_([this]() { cancelled_ = true; }),
```

Comment:
I think you may have to hold "request" object and call its cancel method.  As 

https://github.com/istio/proxy/blob/master/src/envoy/mixer/grpc_transport.cc#L115

Reply:
fixed